### ISSUE TITLE
Setup not working with cli

### ISSUE THREAD

**Issue by Niknox** [Sarjuuk/aowow#34]
I followed your steps until configuration via php cli.

When I issue `php aowow --firstrun` it just shows the following:
```
# php aowow --firstrun
11:49:10          FileGen::init() - creating required directories
11:49:10          created 7 extra paths
11:49:10          Please enter your database credentials.
```
It seems that user interaction is broken for me, it doesn't work with other commands, too.

Output of `php aowow --dbconfig` for example shows:
```
# php aowow --dbconfig
11:53:54          select a numerical index to use the corresponding entry
11:53:54          [0] aowow                  <empty>
11:53:54          [1] world                  <empty>
11:53:54          [2] auth                   <empty>
11:53:54          [3] add an additional Character DB
```

I'm using PHP 5.6.13.

**Comment by Sarjuuk** [comment]
after the first block there should be a prompt 
`Press any key to continue:`

and then the same screen as if using --dbconfig should appear.
it works for me, using PHP 5.5.9-1ubuntu4.11

i'll see if i can get it upgraded, in the meantime i'd suggest you comment the lines in
https://github.com/Sarjuuk/aowow/blob/master/includes/kernel.php#L107 to #L138 and add an
`error_reporting(E_ALL)`
Try again and see what it throws at you.

Also, if you are using WIN, the cli in general is a bit wonky

**Comment by Niknox** [comment]
I'm using Debian 8.

I don't see any prompt appearing, that's why I opened this issue.

Will try your suggestion asap.

**Comment by Niknox** [comment]
Yes I'm missing readline extension for PHP.

Sorry for taking your time, but you might want to add readline to the list of needed extensions.

**Comment by Sarjuuk** [comment]
When you are using WIN, you also have to make do without this extension.
You inputs aren't handled because the script is trying to use a nonexistent function and hits a critical error. You are then writing stuff, but the script that should read it has already terminated. (or at least i think it does, you haven't provided any error message yet)

Try applying this patch, it probably works
https://gist.github.com/Sarjuuk/b88e45d48ac97b0b579a

**Comment by Niknox** [comment]
See above, I'm using Debian. At the time of the last comment I already compiled php with readline support. So the script is working and I get a prompt I just cannot press "Enter". It doesn't seem that the script terminated as I could go back with any signal like "Arrow Down".

I won't apply your patch because readline is there and the functions work properly.

Output of E_ALL: http://pastebin.com/YX5An7xh (nothing important)

**Comment by Sarjuuk** [comment]
hmm, right.

Since i can't reproduce that we'll have to continue that game of guessing :/
Please, try to find out what character is your line terminator:
print_r or echo $keyId [from here](https://github.com/Sarjuuk/aowow/blob/master/setup/tools/CLISetup.class.php#L367)

it should be 10 (\n), but i half expect it to be 13 (\r) (which i originally ignored, because of WIN, which sends both)

**Comment by Niknox** [comment]
Correct, it's 13 (CR). I'm not that much into ascii characters, but isn't it both \r\n in php for new line?

**Comment by Niknox** [comment]
This works for me, thanks for your help!

**Comment by Sarjuuk** [comment]
np

**Comment by Sarjuuk** [comment]
should be fixed with 00bd33abfb8c1c1f9f54091feb6e0e46c4e55e2a

finally :x
